### PR TITLE
SAK-43022: Gradebook: Problem when overrides grades

### PR DESCRIPTION
**Pull request by victorGomollon**
Gradebook: Problem when overrides grades

### CODE REVIEW

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java [hunk 1]
```diff
@@ -3201,25 +3201,30 @@ public Object doInHibernate(final Session session) throws HibernateException {
...
 					// mapped grade
-					final String mappedGrade = GradeMapping.getMappedGrade(sortedGradeMap, calculatedGrade);
-					log.debug("calculatedGrade: {} -> mappedGrade: {}", calculatedGrade, mappedGrade);
+					final String mappedGrade = GradeMapping.getMappedGrade(sortedGradeMap, autoCalculatedGrade);
+					log.debug("calculatedGrade: {} -> mappedGrade: {}", autoCalculatedGrade, mappedGrade);
 					cg.setMappedGrade(mappedGrade);
 
 					// points
-					cg.setPointsEarned(gr.getPointsEarned()); // synonymous with gradeRecord.getCalculatedPointsEarned()
+					cg.setPointsEarned(gr.getPointsEarned()!=null ? (gr.getPointsEarned()*gr.getTotalPointsPossible())/100 : gr.getCalculatedPointsEarned()); // synonymous with gradeRecord.getCalculatedPointsEarned()
```
Comment (ern) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java:3227]: spacing

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java [hunk 2]
```diff
@@ -3201,25 +3201,30 @@ public Object doInHibernate(final Session session) throws HibernateException {
 
 					// calculated grade
 					// may be null if no grade entries to calculate
-					Double calculatedGrade = gr.getAutoCalculatedGrade();
+					Double calculatedGrade = gr.getPointsEarned()!=null ? gr.getPointsEarned() : gr.getAutoCalculatedGrade();
```
Comment (ern) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java:3204]: spacing

File: edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java [hunk 3]
```diff
@@ -33,6 +33,7 @@
 	private static final long serialVersionUID = 1L;
 
 	private String enteredGrade;
+	private Double pointsEarned;
```
Comment (ern) [inline edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java:36]: @Setter and @Getter

**Comment by ern** [comment]
It would be really awesome if we could add a unit test that exercises these calculations
see TestGradebookNgBusinessService

File: edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java (hunk 3 above)
Comment (SedueRey) [reply edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java:36]: get method is already created and we have modify it, that's the reason we didn't add a getter and setter method.

**Review by ern — APPROVED** [review]

**Pull request merged**